### PR TITLE
public profile for 2018-06

### PR DESCRIPTION
**Pull request by sijuman**
*2018-08-04: edited opening post (@joshgav)*

This is the proposed June 2018 profile for public Azure, our first public profile. We will generate packages or other constructs in every SDK which target this group of APIs and versions persistently and stably.

The profile is intended to match services available not only in US Azure but in all other public Azures as well, e.g. China, Germany and US Government. It is also intended to ultimately provide a predictable, stable, supported programming interface for Azure developers.

### CODE REVIEW

**Comment by dsgouda** [comment]
@sijuman please confirm that the resource versions specified in this profile actually exist in the repo specs

File: profile/2018-06-01-profile.json [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+                             "redis",
+                             "checkNameAvailability",
+                             "locations",
+                             "locations/operationResults",
+                             "operations"
+                         ]
+
+        },
+
+        "microsoft.classicCompute": {
```
Comment (marstr) [inline profile/2018-06-01-profile.json]: Without Swagger definitions, these RPs don't mean anything to us. If this is an attempt to provide multiple Compute versions into a profile, it seems like there are some anti-patterns trying to make their way in.

**Review by marstr — CHANGES_REQUESTED** [review]
This isn't my final review, but I think we need to discuss this now.

File: profile/2018-06-01-profile.json (hunk 1 above)
Comment (dsgouda) [reply profile/2018-06-01-profile.json]: What I was trying to say.

Comment (marstr) [reply profile/2018-06-01-profile.json]: To be more constructive about this, the way we would recommend doing this would be to have two profiles.

**Comment by lmazuel** [comment]
@sijuman I realized doing some automatic processing of the file that the structure was incorrect. I allowed myself to push a commit to fix it, to move forward, hope you don't mind :)
@joshgav This means we need schema to check this file now. It was JSON valid, but indented incorrectly.

File: profile/2018-06-01-profile.json (hunk 1 above)
Comment (dsgouda) [reply profile/2018-06-01-profile.json]: @sphibbs FYI

File: profile/2018-06-01-profile.json [hunk 2]
```diff
@@ -0,0 +1,655 @@
...
+                             "locations/CheckDnsNameAvailability",
+                             "locations/operationResults",
+                             "locations/operations",
+                             "locations/usgaes",
+                             "locations/virtualNetworkAvailableEndpointServices",
+                             "networkInterfaces",
+                             "networkSecurityGroups",
+                             "networkWatchers",
+                             "operations",
+                             "piblicIPAddresses",
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Typo

File: profile/2018-06-01-profile.json [hunk 3]
```diff
@@ -0,0 +1,655 @@
...
+                             "ddosProtectionPlans",
+                             "expressRouteCircuits",
+                             "expressRouteServiceProviders",
+                             "loadBalancers",
+                             "localNetworkGateways",
+                             "locations",
+                             "locations/CheckDnsNameAvailability",
+                             "locations/operationResults",
+                             "locations/operations",
+                             "locations/usgaes",
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Typo

File: profile/2018-06-01-profile.json (hunk 1 above)
Comment (lmazuel) [reply profile/2018-06-01-profile.json]: @marstr @dsgouda Microsoft.ClassicCompute is RDFE compat, it's another RP:
https://docs.microsoft.com/en-us/azure/role-based-access-control/resource-provider-operations#microsoftclassiccompute

https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-move-resources#new-subscription

File: profile/2018-06-01-profile.json [hunk 4]
```diff
@@ -0,0 +1,655 @@
...
+                             "operations"
+                         ]
+
+        },
+
+        "microsoft.classicCompute": {
+
+                         "2016-11-01": [
+                             "capabilities",
+                             "checkDomainNameAvailability",
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Typo

File: profile/2018-06-01-profile.json [hunk 5]
```diff
@@ -0,0 +1,655 @@
...
+                         "2016-06-01":[
+                             "locations",
+                             "locations/allocateStamp",
+                             "locations/allocatedStamp",
+                             "locations/backupStatus"
+
+                         ],
+                         
+                         "2016-10-08": [
+                             "operations"
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Based on ARM providers list, this does not exist, but there is:
['2016-08-10', '2016-06-01', '2015-12-15', '2015-12-10', '2015-11-10', '2015-08-15', '2015-08-10', '2015-06-10', '2015-03-15']

File: profile/2018-06-01-profile.json [hunk 6]
```diff
@@ -0,0 +1,655 @@
...
+                             "checkNameAvailability",
+                             "namespaces",
+                             "operations"
+                         ]
+        },
+
+        "microsoft.resourceHealth": {
+
+                         "2015-05-01": [
+                             "operations"
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Based on ARM providers list, this does not exist but 2015-01-01 does

File: profile/2018-06-01-profile.json [hunk 7]
```diff
@@ -0,0 +1,655 @@
...
+                            "locations/virtualNetworkRulesAzureAsyncOperation",
+                            "locations/deleteVirtualNetworkOrSubnetsOperationResults",
+                            "locations/deleteVirtualNetworkOrSubnetsAzureAsyncOperation",
+                            "servers/advisors",
+                            "servers/elasticPools/advisors",
+                            "servers/databases/advisors",
+                            "servers/elasticPoolEstimates",
+                            "servers/databases/auditRecords",
+                            "servers/databases/VulnerabilityAssessmentScans",
+                            "servers/databases/vulnerabiltiyAssessmentSettings",
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Typo

File: profile/2018-06-01-profile.json [hunk 8]
```diff
@@ -0,0 +1,655 @@
...
+                            "servers/databses/syncGroups",
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Typo

File: profile/2018-06-01-profile.json [hunk 9]
```diff
@@ -0,0 +1,655 @@
...
+                            "servers/databases/syncGroups/syncMembers",
+                            "servers/syncAgents",
+                            "locations/syncGroupOperationResults",
+                            "locations/syncMemberOperationResults",
+                            "locations/syncAgentOperationResults",
+                            "locations/syncDtabaseIds"
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Typo

File: profile/2018-06-01-profile.json [hunk 10]
```diff
@@ -0,0 +1,655 @@
...
+                             "servers/extendedAuditingSettings",
+                             "locations/auditingSettingsAzureAsyncOperation",
+                             "locations/auditingSettingsOperationResults",
+                             "location/jobAgentOperationResults",
+                             "locations/jobAgentAzureAsyncOperation",
+                             "servers/dnsAliases",
+                             "locations/dnsAliasAsyncOperation",
+                             "locations/dnsAliasOperationResults",
+                             "locations/databaseRestoreAzureAsyncOperation",
+                             "servers/databses/VulnerabilityAssessment",
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Typo

File: profile/2018-06-01-profile.json [hunk 11]
```diff
@@ -0,0 +1,655 @@
...
+                              "publishingUsers",
+                              "runtimes",
+                              "sites/premieraddons",
+                              "sourceControls"
+
+                          ],
+
+                          "2016-08-01": [
+                            "sites",
+                            "sites/extensions",
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Seems to be sites/instances/extensions

File: profile/2018-06-01-profile.json [hunk 12]
```diff
@@ -0,0 +1,655 @@
...
+                            "sites/hostNameBindings",
+                            "sites/instances",
+                            "sites/instances/extensions",
+                            "sites/slots",
+                            "sites/slots/extensions",
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Seems to be sites/slots/instances/extensions

File: profile/2018-06-01-profile.json [hunk 13]
```diff
@@ -0,0 +1,655 @@
...
+                            "sites/slots/instances",
+                            "sites/slots/instances/extensions"
+                          ],
+
+                          "2016-09-01" : [
+                            "hostingEnvironments",
+                            "hostingEnvironments/multiRolePools",
+                            "hostingEnvironments/multiRolePools/instances",
+                            "hostingEnvironments/workerPools",
+                            "hostingEnvironemts/workPools/instances",
```
Comment (lmazuel) [inline profile/2018-06-01-profile.json]: Typo

File: profile/2018-06-01-profile.json (hunk 11 above)
Comment (sijuman) [reply profile/2018-06-01-profile.json]: Have that one below, it includes both.

File: profile/2018-06-01-profile.json (hunk 12 above)
Comment (sijuman) [reply profile/2018-06-01-profile.json]: Have that one below as well, both of these exist according to my list

Comment (lmazuel) [reply profile/2018-06-01-profile.json]: Based on public Azure provider list, it doesn't exist.

Comment (sijuman) [reply profile/2018-06-01-profile.json]: The Azure provider list provided to me by Josh has both. I can reach out to service teams and see, but if there is no swagger I can delete, just let me know.

Comment (lmazuel) [reply profile/2018-06-01-profile.json]: FYI @joshgav

**Comment by salameer** [comment]
@sijuman what's the deal with this profile, should it be in should it not?

I will close it since there's no activity for a while now. please request reopening through email if still required

**Pull request closed without merging**